client/src/components/PriceVolumeChart.jsx

```diff
@@ -34,25 +41,66 @@ class PriceVolumeChart extends React.Component {
                 heighest: data[0].heighest,
                 averagePrice: data[0].averagePrice,
                 currentPrice: data[0].currentPrice,
-                difference: diff
+                difference: diff,
+                currentPriceIndex: currentPriceIndex,
+                averagePriceIndex: averagePriceIndex 
             });
         })
     }
+
     getDifference(averagePrice, currentPrice){
         return Math.round(currentPrice/averagePrice*100-100);
     }
+
+    findBarGraphIndex(price, prices){
+        let left = 0;
+        let right = prices.length - 1;
+        let mid = 0;
+        while(left<=right){
+            mid = Math.round((left+right)/2);
+            if(prices[mid] === price) return mid;
+            else if(prices[mid] < price){
+                left = mid+1;
+            }else{
+                right = mid-1;
+            }
+        }
+        return mid;
+    }
+
+    indexInRange(i){
+        let index1 = this.state.currentPriceIndex;
+        let index2 = this.state.averagePriceIndex;
+        if(index1<index2){
+            return (index1<=i) && (i<=index2);
+        }
+        return (index2<=i) && (i<=index1);
+    }
+
     render(){
         return (
             <div>
                 <h1>Price Paid on Robinhood</h1>
-                <h2>Name: <span id='name'>{this.state.name}</span></h2>
-                <h2>52 Week Low: $<span id='lowest'>{this.state.lowest}</span></h2>
-                <h2>52 Week High: $<span id='heighest'>{this.state.heighest}</span></h2>
-                <h2>Average Price Paid: $<span id='average'>{this.state.averagePrice.toFixed(2)}</span></h2>
-                <h2>Current Price: $<span id='current'>{this.state.currentPrice}</span></h2>
-                <h2 id='difference'><span>{Math.abs(this.state.difference)}</span>% {this.state.difference>0 ? 'Higher':'Lower'}</h2>
+                <svg viewBox="0 0 676 200" width="676" height="200">
+                    <rect x="0" y="0" width="676" height="200" id="bargraph" fill="#1b1b1d"></rect>
+                    <g transform="translate(0,180) scale(1,-1)">
+                        {
+                            this.state.volumes.map((h, i)=>{
+                                return(
+                                <rect rx="1" x={this.barWidth*2*i} y="0" width={this.barWidth} height={h} fill={this.indexInRange(i) ? "#20ce99" : "#0e0d0d"}>
```

Comment:
can use react self closing tags for < rect /> for cleaner code and maybe indent after return ?